### ISSUE TITLE
Spurious BYOND topic socket closures

### ISSUE THREAD

**Issue by Cyberboss** [tgstation/tgstation-server#1529]
Affects mainly CI, but could cause issues in production too.

Starting over after the BYOND.TopicSender changes:

Seen at 
```
65295
65487
65153
65528
65530
```

Windows:
```
2023-06-08T08:42:54.1477538Z  Topic recieve limit test S:65297...
2023-06-08T08:42:54.1478046Z  [6/8/2023 8:34:38 AM +00:00] TEST ERROR: System.Net.Sockets.SocketException (10054): An existing connection was forcibly closed by the remote host.
2023-06-08T08:42:54.1479064Z     at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
2023-06-08T08:42:54.1480024Z     at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
2023-06-08T08:42:54.1480841Z     at System.Threading.Tasks.ValueTask`1.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
2023-06-08T08:42:54.1481929Z  --- End of stack trace from previous location ---
2023-06-08T08:42:54.1482608Z     at System.Threading.Tasks.TaskToApm.End[TResult](IAsyncResult asyncResult)
2023-06-08T08:42:54.1483245Z     at Byond.TopicSender.TopicClient.<>c__DisplayClass2_0`1.<AsyncSocketOperation>b__0(IAsyncResult asyncResult)
2023-06-08T08:42:54.1483919Z  --- End of stack trace from previous location ---
2023-06-08T08:42:54.1484556Z     at Byond.TopicSender.TopicClient.AsyncSocketOperation[TResult](Func`2 start, Func`2 end, TimeSpan timeout, CancellationToken cancellationToken)
2023-06-08T08:42:54.1485354Z     at Byond.TopicSender.TopicClient.SendTopic(IPEndPoint endPoint, String queryString, CancellationToken cancellationToken)
2023-06-08T08:42:54.1486369Z     at Tgstation.Server.Tests.Live.Instance.WatchdogTest.ValidateTopicLimits(CancellationToken cancellationToken) in D:\a\tgstation-server\tgstation-server\tests\Tgstation.Server.Tests\Live\Instance\WatchdogTest.cs:line 593
2023-06-08T08:42:54.1487790Z     at Tgstation.Server.Tests.Live.Instance.WatchdogTest.InteropTestsForLongRunningDme(CancellationToken cancellationToken) in D:\a\tgstation-server\tgstation-server\tests\Tgstation.Server.Tests\Live\Instance\WatchdogTest.cs:line 110
2023-06-08T08:42:54.1488994Z     at Tgstation.Server.Tests.Live.Instance.WatchdogTest.Run(CancellationToken cancellationToken) in D:\a\tgstation-server\tgstation-server\tests\Tgstation.Server.Tests\Live\Instance\WatchdogTest.cs:line 96
2023-06-08T08:42:54.1490155Z     at Tgstation.Server.Tests.Live.Instance.InstanceTest.RunTests(CancellationToken cancellationToken, Boolean highPrioDD, Boolean lowPrioDeployment) in D:\a\tgstation-server\tgstation-server\tests\Tgstation.Server.Tests\Live\Instance\InstanceTest.cs:line 51
```

**Comment by Cyberboss** [comment]
No size correlation unfortunately it seems
```
2023-06-09T18:29:33.6250846Z  Topic recieve limit test S:65281...
2023-06-09T18:29:33.6251419Z  [6/9/2023 6:21:37 PM +00:00] TEST ERROR: System.Net.Sockets.SocketException (10054): An existing connection was forcibly closed by the remote host.
2023-06-09T18:29:33.6252317Z     at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.ThrowException(SocketError error, CancellationToken cancellationToken)
2023-06-09T18:29:33.6253528Z     at System.Net.Sockets.Socket.AwaitableSocketAsyncEventArgs.System.Threading.Tasks.Sources.IValueTaskSource<System.Int32>.GetResult(Int16 token)
2023-06-09T18:29:33.6254455Z     at System.Threading.Tasks.ValueTask`1.ValueTaskSourceAsTask.<>c.<.cctor>b__4_0(Object state)
2023-06-09T18:29:33.6255306Z  --- End of stack trace from previous location ---
2023-06-09T18:29:33.6255855Z     at System.Threading.Tasks.TaskToApm.End[TResult](IAsyncResult asyncResult)
2023-06-09T18:29:33.6256514Z     at Byond.TopicSender.TopicClient.<>c__DisplayClass2_0`1.<AsyncSocketOperation>b__0(IAsyncResult asyncResult)
2023-06-09T18:29:33.6257186Z  --- End of stack trace from previous location ---
2023-06-09T18:29:33.6259134Z     at Byond.TopicSender.TopicClient.AsyncSocketOperation[TResult](Func`2 start, Func`2 end, TimeSpan timeout, CancellationToken cancellationToken)
2023-06-09T18:29:33.6260751Z     at Byond.TopicSender.TopicClient.SendTopic(IPEndPoint endPoint, String queryString, CancellationToken cancellationToken)
2023-06-09T18:29:33.6262551Z     at Tgstation.Server.Tests.Live.Instance.WatchdogTest.ValidateTopicLimits(CancellationToken cancellationToken) in D:\a\tgstation-server\tgstation-server\tests\Tgstation.Server.Tests\Live\Instance\WatchdogTest.cs:line 594
2023-06-09T18:29:33.6264407Z     at Tgstation.Server.Tests.Live.Instance.WatchdogTest.InteropTestsForLongRunningDme(CancellationToken cancellationToken) in D:\a\tgstation-server\tgstation-server\tests\Tgstation.Server.Tests\Live\Instance\WatchdogTest.cs:line 110
2023-06-09T18:29:33.6265952Z     at Tgstation.Server.Tests.Live.Instance.WatchdogTest.Run(CancellationToken cancellationToken) in D:\a\tgstation-server\tgstation-server\tests\Tgstation.Server.Tests\Live\Instance\WatchdogTest.cs:line 96
2023-06-09T18:29:33.6267523Z     at Tgstation.Server.Tests.Live.Instance.InstanceTest.RunTests(CancellationToken cancellationToken, Boolean highPrioDD, Boolean lowPrioDeployment) in D:\a\tgstation-server\tgstation-server\tests\Tgstation.Server.Tests\Live\Instance\InstanceTest.cs:line 51
2023-06-09T18:29:33.6268987Z     at Tgstation.Server.Tests.Live.TestLiveServer.<>c__DisplayClass20_0.<<TestTgsInternal>g__FailFast|2>d.MoveNext() in D:\a\tgstation-server\tgstation-server\tests\Tgstation.Server.Tests\Live\TestLiveServer.cs:line 879
```

**Comment by Cyberboss** [comment]
Better but not entirely fixed

**Comment by Cyberboss** [comment]
This might just be BYOND fail2banning us

**Comment by Cyberboss** [comment]
Probably fixed with bd80c592babdf26a028140a8ed9366a24bcc2f2a
